pgjdbc/src/main/java/org/postgresql/jdbc/PgClob.java

```diff
@@ -20,60 +20,110 @@ public PgClob(org.postgresql.core.BaseConnection conn, long oid) throws java.sql
     super(conn, oid);
   }
 
-  public synchronized Reader getCharacterStream(long pos, long length) throws SQLException {
-    checkFreed();
-    throw org.postgresql.Driver.notImplemented(this.getClass(), "getCharacterStream(long, long)");
+  public Reader getCharacterStream(long pos, long length) throws SQLException {
+    lock.lock();
+    try {
+      checkFreed();
+      throw org.postgresql.Driver.notImplemented(this.getClass(), "getCharacterStream(long, long)");
+    } finally {
+      lock.unlock();
+    }
   }
 
-  public synchronized int setString(long pos, String str) throws SQLException {
-    checkFreed();
-    throw org.postgresql.Driver.notImplemented(this.getClass(), "setString(long,str)");
+  public int setString(long pos, String str) throws SQLException {
+    lock.lock();
+    try {
+      checkFreed();
+      throw org.postgresql.Driver.notImplemented(this.getClass(), "setString(long,str)");
+    } finally {
+      lock.unlock();
+    }
   }
 
-  public synchronized int setString(long pos, String str, int offset, int len) throws SQLException {
-    checkFreed();
-    throw org.postgresql.Driver.notImplemented(this.getClass(), "setString(long,String,int,int)");
+  public int setString(long pos, String str, int offset, int len) throws SQLException {
+    lock.lock();
+    try {
+      checkFreed();
+      throw org.postgresql.Driver.notImplemented(this.getClass(), "setString(long,String,int,int)");
+    } finally {
+      lock.unlock();
+    }
   }
 
-  public synchronized java.io.OutputStream setAsciiStream(long pos) throws SQLException {
-    checkFreed();
-    throw org.postgresql.Driver.notImplemented(this.getClass(), "setAsciiStream(long)");
+  public java.io.OutputStream setAsciiStream(long pos) throws SQLException {
+    lock.lock();
+    try {
+      checkFreed();
+      throw org.postgresql.Driver.notImplemented(this.getClass(), "setAsciiStream(long)");
+    } finally {
+      lock.unlock();
+    }
   }
 
-  public synchronized java.io.Writer setCharacterStream(long pos) throws SQLException {
-    checkFreed();
-    throw org.postgresql.Driver.notImplemented(this.getClass(), "setCharacteStream(long)");
+  public java.io.Writer setCharacterStream(long pos) throws SQLException {
+    lock.lock();
+    try {
+      checkFreed();
+      throw org.postgresql.Driver.notImplemented(this.getClass(), "setCharacteStream(long)");
+    } finally {
+      lock.unlock();
+    }
   }
 
-  public synchronized InputStream getAsciiStream() throws SQLException {
-    return getBinaryStream();
+  public InputStream getAsciiStream() throws SQLException {
+    lock.lock();
+    try {
+      return getBinaryStream();
```

Comment:
is getBinaryStream() synchronized appropriately? It seems synchronization here is not needed.